addons/website_sale/models/sale_order.py

```diff
@@ -55,15 +55,19 @@ def _compute_abandoned_cart(self):
                 order.is_abandoned_cart = False
 
     def _search_abandoned_cart(self, operator, value):
-        abandoned_delay = self.website_id and self.website_id.cart_abandoned_delay or 1.0
-        abandoned_datetime = fields.Datetime.to_string(datetime.utcnow() - relativedelta(hours=abandoned_delay))
-        abandoned_domain = expression.normalize_domain([
-            ('date_order', '<=', abandoned_datetime),
-            ('website_id', '!=', False),
+        website_ids = self.env['website'].search_read(fields=['id', 'cart_abandoned_delay', 'partner_id'])
```

Comment:
@fmdl 

Do you want to validate it before we merge ?
Imo, It should fix your problems without extra raw sql

//cc @nim-odoo

Reply:
//Cc @sswapnesh ... we already discuss a lot about it, so you can be interested too